src/test/java/cloud/fogbow/common/plugins/cloudidp/opennebula/OpenNebulaIdentityProviderPluginTest.java

```diff
@@ -0,0 +1,123 @@
+import org.opennebula.client.user.UserPool;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.HashMap;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({OpenNebulaClientUtil.class})
+public class OpenNebulaIdentityProviderPluginTest {
+
+    private OpenNebulaIdentityProviderPlugin plugin;
+    public static final String FAKE_IDENTITY_URL = "http://localhost:0000";
+    public static final String FAKE_USER_ID = "fake-id";
+    public static final String FAKE_USER_PASSWORD = "fake-password";
+    public static final String INVALID_PASSWORD = "invalid-password";
+    public static final String ONE_RESPONSE_MESSAGE = "Random oneResponse return message";
+
+    @Before
+    public void setup() throws UnexpectedException {
+        this.plugin = Mockito.spy(new OpenNebulaIdentityProviderPlugin(FAKE_IDENTITY_URL));
+
+        Client client = Mockito.mock(Client.class);
+
+        PowerMockito.mockStatic(OpenNebulaClientUtil.class);
+        BDDMockito.given(OpenNebulaClientUtil.createClient(Mockito.anyString(), Mockito.anyString())).willReturn(client);
+    }
+
+    @Test
+    public void testAuthenticateSuccess() throws FogbowException {
+        // setup
+        Mockito.doReturn(true).when(this.plugin).isAuthenticated(Mockito.anyString());
+        HashMap<String,String> credentials = getCredentials(FAKE_USER_ID, FAKE_USER_PASSWORD);
+
+        // exercise
+        OpenNebulaUser openNebulaUser = this.plugin.getCloudUser(credentials);
+
+        // verify
+        Assert.assertNotNull(openNebulaUser);
+        Assert.assertEquals(FAKE_USER_ID, openNebulaUser.getId());
+        Mockito.verify(this.plugin).isAuthenticated(Mockito.anyString());
+    }
+
+    @Test(expected = UnauthenticatedUserException.class)
+    public void testAuthenticateUnsuccess() throws FogbowException {
+        Mockito.doReturn(false).when(this.plugin).isAuthenticated(Mockito.anyString());
+        HashMap<String,String> credentials = getCredentials(FAKE_USER_ID, INVALID_PASSWORD);
+
+        // exercise
+        OpenNebulaUser openNebulaUser = this.plugin.getCloudUser(credentials);
+    }
+
+    @Test(expected = InvalidParameterException.class)
+    public void testAuthenticateWithNullParams() throws FogbowException {
+        HashMap<String,String> credentials = getCredentials(FAKE_USER_ID, null);
+
+        // exercise
+        OpenNebulaUser openNebulaUser = this.plugin.getCloudUser(credentials);
+    }
+
+    @Test(expected = InvalidParameterException.class)
+    public void testAuthenticateWithEmptyParams() throws FogbowException {
+        HashMap<String,String> credentials = getCredentials(FAKE_USER_ID, "");
+
+        // exercise
+        OpenNebulaUser openNebulaUser = this.plugin.getCloudUser(credentials);
+    }
+
+    @Test
+    public void testIsAuthenticatedSuccessWithoutMockingHelperMethod() throws FogbowException {
```

Comment:
why do we need this testcase?

Reply:
To make sure the helper method behaves appropriately.